libcontainer/configs/device.go

```diff
@@ -35,23 +28,153 @@ type Device struct {
 
 	// Gid of the device.
 	Gid uint32 `json:"gid"`
+}
 
-	// Write the file to the allowed list
-	Allow bool `json:"allow"`
+// DevicePermissions is a cgroupv1-style string to represent device access. It
+// has to be a string for backward compatibility reasons, hence why it has
+// methods to do set operations.
+type DevicePermissions string
+
+const (
+	deviceRead uint = (1 << iota)
+	deviceWrite
+	deviceMknod
+)
+
+func (p DevicePermissions) toSet() uint {
+	var set uint
+	if strings.Contains(string(p), "r") {
+		set |= deviceRead
+	}
+	if strings.Contains(string(p), "w") {
+		set |= deviceWrite
+	}
+	if strings.Contains(string(p), "m") {
+		set |= deviceMknod
+	}
```

Comment:
1. Is it OK to ignore unknown characters here?

2. I would rewrite it to do
```go
for _, c := range string(p) {
  switch c {
  case 'r':
    set |= deviceRead
  case 'w':
......
....
```

Reply:
> Is it OK to ignore unknown characters here?

oh, I see you have `IsValid()` below, nice.

Reply:
Yeah, I'll switch this to also use the iterator setup.